test/jasmine/tests/legend_scroll_test.js

```diff
@@ -88,10 +92,28 @@ describe('The legend', function() {
                 done();
             });
         });
+        
+        it('should resize when relayout\'ed with new height', function(done) {
+            var origLegendHeight = getBBox(legend).height;
+            
+            Plotly.relayout(gd, {'height':gd._fullLayout.height/2}).then(function() {
+                var legend = document.getElementsByClassName('legend')[0];
+                var legendHeight = getBBox(legend).height;
+                //legend still exists and not duplicated
+                expect(countLegendGroups(gd)).toBe(1);
+                expect(countLegendClipPaths(gd)).toBe(1);
+                
+                // clippath resized to new height less than new plot height
+                expect(+legendHeight).toBe(getPlotHeight(gd));
+                expect(+legendHeight).toBeLessThan(+origLegendHeight);
```

Comment:
Could you check against the computed height using `toBeCloseTo`?

It helps catch potential bugs earlier than just relative size comparison.